### PR TITLE
Getting-Started-With-Web-Scraping-Using-Python

### LINKED ISSUE #455: Getting started with web scraping using Python
### Brief Summary: 

Web scraping is a technique in which a computer program extracts data from human-readable output coming from a website.
- **quetext** website which you are using to check our articles for plagiarism is using web scraping.
- flights & hotels booking websites are using it in order to know the lowest price for a ticket, etc.
- a lot of companies are scraping social network profiles in order to determine the posts with the most interactions. then to create strategic marketing decisions.

In this tutorial, I will be using python with **beautiful soup** library to make the reader understand what is web scraping, how he can use it to gather important information in a good format, and we will create two programs step by step, one of them will be a Covid19 Desktop Notifier which makes a live update on COVID-19 Cases and Deaths for your country using a popup notification on desktop.

### Key Takeaways: 

 - what is a web scraping and how it is working.
 - how to check if the website is allowing us to do web scraping, in order to make the operation legal(80-90% of the websites are allowing web scraping)
- getting started with **beautiful soap** library and plyer.
- practicing web scraping using Python in two programs.

### References: 

N/A

### PR DESCRIPTION
**Pull request by ahmadmardeni1**
Note: I designed the hero image myself but I'm not sure if it is accepted, please feel free asking me to do another design if you didn't like it.

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      

#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tool:** [Grammarly](https://grammarly.com)
- [ ] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

this Fixes #455

### CODE REVIEW

**Comment by ahmadmardeni1** [comment]
Fixes #455

**Comment by hectorkambow** [comment]
Plagiarism check done

**Comment by hectorkambow** [comment]
👋  Good morning @AskAlice  
Happy Friday. 
Wanted to ask if you could give this "Web-Scraping-Using-Python" article a tech review whenever you have a chance before we publish. 
Thank you v much 🚀

File: articles/getting-started-with-web-scraping-using-python/index.md [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+```
+sudo apt-get install bs4 urllib3 plyer requests
+```
+
+You are ready to go, let's get started and learn a bit more on **web scraping** through two real-life projects.
+
+### Reddit Web Scraper
+One year ago, I wanted to build a smart **AI** bot, my aim was to make it talk like a human, but I had a problem, I didn't have a good dataset to train my bot on, so I decided to use **REDDIT** posts and comments.
+
+Here we will go through how to build the basics of that app step by step, and we will use https://old.reddit.com/
```
[the remark this answers is not in the document]
Comment (ahmadmardeni1) [reply articles/getting-started-with-web-scraping-using-python/index.md:101]: done @hectorkambow

File: articles/getting-started-with-web-scraping-using-python/index.md [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+
+Finally, we did a get request in order to connect to that URL then to pull out the HTML code for that page using the **Beautiful Soup** library.
+
+Now let's move on to the next step of building our app:
+
+Open this [URL](https://old.reddit.com/top/) then press **F12** to inspect the page, you will see the HTML code for it. In order to know in what line you can find the HTML code for the element you want to locate, you have to do a right-click on that element then click on inspect.
+
+After doing the process above on the first title on the page, you can see the following code with a highlight for the tag which holds the data you right-clicked on:
+
+![](/engineering-education/getting-started-with-web-scraping-using-python/1.PNG)
```
[the remark this answers is not in the document]
Comment (ahmadmardeni1) [reply articles/getting-started-with-web-scraping-using-python/index.md:130]: I'm confused, what should I change here?? @AskAlice

**Comment by ahmadmardeni1** [comment]
good afternoon @hectorkambow 🚀
I included the information that Alice told me about.
also, I appreciate the minor changes you did but I changed some grammar mistakes and I'm a little confused about why you deleted the Reddit Web Scraper example in the conclusion and leaving its line empty.

**Comment by hectorkambow** [comment]
> good afternoon @hectorkambow 🚀
> I included the information that Alice told me about.
> also, I appreciate the minor changes you did but I'm a little confused about why you deleted the Reddit Web Scraper example in the conclusion and leaving its line empty.

Good morning @ahmadmardeni1 
Thank you for including the feedback from Alice. 

I believe you are referring to this line: "consider that a specific post receives a lot of comments but without any like, we can know as a result that the post strikes a chord in people and go against the community."

I removed the sentence - because it wasn't quite clear what it was showcasing. It wasn't very clear how that sentence tied back in - I was not aware it was an example. 

We can add it back in - if we elaborate it a bit - make it very clear that it is an example. 👍

**Comment by ahmadmardeni1** [comment]
Thank you for your explanation 🚀 @hectorkambow 
I changed that sentence and I hope it is clear now.

**Pull request merged**